algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/molecular/ProteinIsWildTypeByIHC.kt

```diff
@@ -4,20 +4,23 @@ import com.hartwig.actin.algo.evaluation.EvaluationFactory
 import com.hartwig.actin.algo.evaluation.EvaluationFunction
 import com.hartwig.actin.datamodel.PatientRecord
 import com.hartwig.actin.datamodel.algo.Evaluation
+import java.time.LocalDate
 
-class ProteinIsWildTypeByIHC(private val protein: String) : EvaluationFunction {
+class ProteinIsWildTypeByIHC(private val protein: String, private val gene: String, private val maxTestAge: LocalDate? = null) : EvaluationFunction {
 
     override fun evaluate(record: PatientRecord): Evaluation {
         val allIHCTestsForProtein =
             IhcTestFilter.allIHCTestsForProtein(record.priorIHCTests, protein)
         val hasOnlyWildTypeResults = allIHCTestsForProtein.isNotEmpty() && allIHCTestsForProtein.all { test ->
             WILD_TYPE_QUERY_STRINGS.any { it.equals(test.scoreText, ignoreCase = true) }
         }
+        val geneIsWildType = MolecularRuleEvaluator.geneIsWildTypeForPatient(gene, record, maxTestAge)
+        val additionalMessage = if (geneIsWildType) " though $gene is wild-type in recent molecular test" else ""
```

Comment:
Minor but you could consider moving these steps into the else since they are only used there.

And now that I copy-paste this 3x, it may be nice to place this logic into a separate function (you'd want the same `additionalMessage` in the end for every situation